File: Transportation-model/src/main/java/com/transportation/transportation/model/entities/StationSuggestion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.transportation.transportation.model.entities;

import org.codehaus.jackson.annotate.JsonSubTypes;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 *
 * @author youssef
 */
@Document(collection = "stationSuggestion")
@JsonSubTypes({
    @JsonSubTypes.Type(BusStationSuggestion.class),
    @JsonSubTypes.Type(TrainStationSuggestion.class),
    @JsonSubTypes.Type(TramwayStationSuggestion.class)
})
public class StationSuggestion extends Station {

    public StationSuggestion() {
    }

    public StationSuggestion(String id) {
        super(id);
    }

}

File: DartWebClient/lib/model/station_suggestion.dart
part of station;

class StationSuggestion extends Station {
  StationSuggestion(num lat, num lng, [String id]) : super(lat, lng, id);
}

class BusStationSuggestion extends StationSuggestion {
  BusStationSuggestion(num lat, num lng, [String id]) : super(lat, lng, id) {
  }
}

class TrainStationSuggestion extends StationSuggestion {
  TrainStationSuggestion(num lat, num lng, [String id]) : super(lat, lng, id) {
  }
}

class TramwayStationSuggestion extends StationSuggestion {
  TramwayStationSuggestion(num lat, num lng, [String id]) : super(lat, lng, id) {
  }
}
